nengo_dl/transform_builders.py

```diff
@@ -89,7 +115,7 @@ def build_pre(self, signals, config):
             perm_x[1:-1] = perm_x[2:]
             perm_x[-1] = 1
 
-            self.perm_x = tf.constant(perm_x)
+            self.perm_x = perm_x
```

Comment:
It's better to do the `tf.constant` here in `build_pre` (as that is just called once). Otherwise the np array will be converted to a tensorflow array every time the op is applied in `build_step` (which can be multiple times, e.g. once for inference and once for training, or every time the simulation step is unrolled).